//Copyright 2024 Sonardyne

//Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
//documentation files (the “Software”), to deal in the Software without restriction, including without limitation the
//rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
//permit persons to whom the Software is furnished to do so, subject to the following conditions:

//The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
//Software.

//THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
//COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
//OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

syntax = "proto3";

import "sonardyne-public/idl/common/result.proto";
import "sonardyne-public/idl/common/timestamp.proto";

import "google/protobuf/any.proto";

package sonardyne.api.pub.configuration;

// A container of instrument configurations, a timestamp and a result.
message ConfigurationEnvelope{
  sonardyne.api.pub.common.Timestamp timestamp = 1; // Timestamp of the configuration
  sonardyne.api.pub.common.Result result = 2;       // Populated by instrument when replying to a SetState indicating the success of the operation
  repeated google.protobuf.Any configuration = 3;   // The configuration data
}

message ConfigurationRequest{
  sonardyne.api.pub.common.Timestamp timestamp = 1; //  Timestamp of the request
  string requestor = 2;                             // Optional user-definable field
}








